hardware/raspberrypi/boot_diagnostics.md

```diff
@@ -1,6 +1,6 @@
 ## Boot Diagnostics on the Raspberry Pi 4
 
-The bootloader on the Raspberry Pi 4 is stored in EEPROM, and as well as providing boot services, is also capable of providing diagnostic information on an attached HDMI display. To display this information, power down the Raspberry Pi 4, remove the SD card and power back up. A diagnostic display similar to below should appear on the attached display.
+The bootloader on the Raspberry Pi 4 is stored in EEPROM, and as well as providing boot services, versions newer than those dated 2020-04-16 are also capable of providing diagnostic information on an attached HDMI display. To display this information, power down the Raspberry Pi 4, remove the SD card and power back up. A diagnostic display similar to below should appear on the attached display.
```

Comment:
Is it worth linking "newer than those dated 2020-04-16" to the EEPROM-update instructions page?

Reply:
Is there anything on there that needs the dating? Didn't see anything with a quick read through.

Reply:
Sorry, I meant that maybe it'd be worth linking **from** this page **to** the eeprom-update-instructions page, for the benefit of somebody who might be reading this page and thinking "oooh, I want some HDMI diagnostics action, how can I update to a newer bootloader?" :wink: 

Or does @timg236 's comment above about the 2020-04-16 release being a "critical" update mean that most users will be updated to a new-enough bootloader version automatically, via `apt upgrade` ?

Reply:
Comma splice: the highlighted insertion (from 'versions') is a new sentence, so a full stop before it.

Reply:
> Or does @timg236 's comment above about the 2020-04-16 release being a "critical" update mean that most users will be updated to a new-enough bootloader version automatically, via `sudo apt upgrade` ?

Similar issues with previous changes have resulted in the docs specifying the first version/date of a product which have a particular feature (or lack of problem). I think this one is worth documenting.

Reply:
> Comma splice: the highlighted insertion (from 'versions') is a new sentence, so a full stop before it.

Putting a full stop before 'versions' would change the first sentence to "The bootloader on the Raspberry Pi 4 is stored in EEPROM, and as well as providing boot services." which wouldn't make sense? The current sentence is a bit long, but I think it's okay as-is.

Reply:
> > Comma splice: the highlighted insertion (from 'versions') is a new sentence, so a full stop before it.
> 
> Putting a full stop before 'versions' would change the first sentence to "The bootloader on the Raspberry Pi 4 is stored in EEPROM, and as well as providing boot services." which wouldn't make sense? The current sentence is a bit long, but I think it's okay as-is.

Oops. Lochdown must be getting to me - that's mistake 3 today - doh! The current sentence is not gramatically correct - there's still comma splice in there where I said. Whether this matters or not I'm not sure, but I tend to take the view that it is best not to introduce more errors.

Reply:
Missed the thread (whilst debugging some brainbending XHCI stuff). Unless anyone scream this week the current 'stable' release 2020-04-16 will be pushed out as a critical update and made the default as Pi Imager.
When I say push, it would be to say that it's made available and will be automatically installed  after an apt-upgrade so it's really pull compared to things like phone updates.

Reply:
>made the default as Pi Imager.

Presumably you mean it will be made the current 'critical' firmware release, and made the recovery image available via the Raspberry Pi Imager tool?

Reply:
Yes, Pi Imager recover option will bring in this new bootloader. And it will also be in apt full-upgrade.